src/components/BaseSeparatePage/TableOfContents/useHeadingsData.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react'
+
+/**
+ * Dynamically generates the table of contents list
+ */
+
+export type NestedHeadingtype = {
+  id: string
+  title: string
+  items?: NestedHeadingtype[]
+}
+
+const useHeadingsData = () => {
```

Comment:
This implementation has too much implicit coupling and is only globally singleton, requiring refactoring